### PR TITLE
Attr error (bsc#1067320)

### PR DESCRIPTION
**Pull request by teclator**
- Hope to fix the last issue with the Firewall & SSH proposal, covered now by tests.

### CODE REVIEW

File: test/proposal_settings_test.rb [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    it "sets the ssh port to be closed" do
+      subject.open_ssh!
+      expect(subject.open_ssh).to be(true)
+      subject.close_ssh!
+      expect(subject.open_ssh).to be(false)
+    end
+  end
+
+  describe "#open_vnc!" do
+    it "sets the vnc port to be opened" do
```
Comment (mchf) [inline test/proposal_settings_test.rb]: most probably copy - paste issue here

File: test/proposal_settings_test.rb [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      subject.disable_sshd!
+    end
+  end
+
+  describe "#open_ssh!" do
+    it "sets the ssh port to be opened" do
+      subject.close_ssh!
+      expect(subject.open_ssh).to be(false)
+      subject.open_ssh!
+      expect(subject.open_ssh).to be(true)
```
Comment (mchf) [inline test/proposal_settings_test.rb]: hugh. I don't understand the code well -> most probably will need to go deeply. 

I'm guessing here, but is ```open_ssh``` just a status checker? Shouldn't it be ```opened_ssh?``` instead

argh it is attribute accessor. So, 1) shouldn't be attr_reader enough for ```open_ssh```? Is breaking the encapsulation really needed? 2) this workflow ```close_ssh!``` -> ```open_ssh``` --[port remains closed]--> ```open_ssh!``` --[port finally opened] looks a little bit weird at first view.

File: src/lib/y2firewall/proposal_settings.rb [hunk 3]
```diff
@@ -68,14 +68,14 @@ def open_vnc!
...
-      self.close_vnc = true
+      self.open_vnc = false
     end
 
   private
 
     def load_feature(feature, to, source: global_section)
       value = Yast::Ops.get(source, feature.to_s)
-      self.send("#{to}=", value) unless value.nil?
+      send("#{to}=", value) unless value.nil?
```
Comment (jreidinger) [inline src/lib/y2firewall/proposal_settings.rb]: public send is not enough? readers are public.

File: test/proposal_settings_test.rb [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+      subject.disable_firewall!
+    end
+  end
+
+  describe "#enable_sshd!" do
+    it "sets sshd service to be enabled" do
+      allow(Yast::PackagesProposal).to receive("AddResolvables")
+        .with("firewall", :package, ["openssh"])
+
+      subject.enable_sshd!
```
Comment (jreidinger) [inline test/proposal_settings_test.rb]: when seeing this, what is reason to have these methods with "!" ?

Comment (teclator) [reply test/proposal_settings_test.rb]: well if it changes the internal state of the object then the convention is to use "!" isn't it?

File: test/proposal_settings_test.rb (hunk 2 above)
Comment (teclator) [reply test/proposal_settings_test.rb]: I could to initializing the attributes, but also considered that it abstract from initialization and you at the end test that both methods are the opposite but as you prefer

File: test/proposal_settings_test.rb (hunk 4 above)
Comment (jreidinger) [reply test/proposal_settings_test.rb]: well, see discussion at https://stackoverflow.com/questions/7179016/what-is-the-purpose-of-and-at-the-end-of-method-names
If it is surprising or dangerous operation it should have "!". For me it indicates like it it try to enable it now and can raise if failed ( but maybe just my POV, we can ask someone else how he see it ). @mvidner @lslezak or @imobachgs ?

Comment (teclator) [reply test/proposal_settings_test.rb]: I'm fine with the discussion. As this implementation was mainly for the POC, what if we maintain the discussion and maybe change the design but merge this PR that is now covering with unit tests the code and then apply the discussion results in next PR.

Are you fine with that?

Comment (teclator) [reply test/proposal_settings_test.rb]: And well about surprising or dangerous, it is not always the case: usually the exclamation mark (!)  means the object is modified like with `downcase!` `sort!` and so on while `downcase` `sort`don't modify the object.

**Pull request merged**